src/meshdb/utils/spreadsheet_import/parse_node.py

```diff
@@ -106,24 +105,19 @@ def normalize_building_node_links(building: models.Building, node: models.Node):
         for n in b.nodes.all():
             nodes_in_cluster.add(n)
 
-        if b.primary_node:
-            established_primary_nodes.add(b.primary_node)
+    active_nodes = [node for node in nodes_in_cluster if node.status == node.NodeStatus.ACTIVE]
 
-    established_primary_nodes = sorted(list(established_primary_nodes), key=lambda node: node.install_date)
+    primary_node = [
+        node
+        for node in sorted(
+            active_nodes if active_nodes else nodes_in_cluster,
+            key=lambda node: node.install_date,
+        )
+    ][0]
 
     for b in buildings_in_cluster:
         for n in nodes_in_cluster:
             b.nodes.add(n)
 
-        if len(established_primary_nodes):
-            if len(established_primary_nodes) > 1:
-                logging.error(
-                    f"Detected multiple primary nodes in the same cluster: "
-                    f"{established_primary_nodes}. These will be consolidated "
-                    f"to the oldest node: {established_primary_nodes[0]}."
-                )
-            b.primary_node = established_primary_nodes[0]
-        else:
-            b.primary_node = node
-
+        b.primary_node = primary_node
```

Comment:
Nit: Why not just set `b.primary_node` directly above on line 110?

Reply:
Because we have to set it for each building in `buildings_in_cluster`? I'm not sure what you're getting at. Do you want me to move the logic on 110 inside this for loop?

Reply:
Oh wait I am foolish. NVM.